### DISCUSSION TITLE
Give TcpListener.AcceptXAsync a CancellationToken

### DISCUSSION THREAD

**Discussion by JC3** [dotnet/runtime#53254] [Ideas]
It would be great if `TcpListener.AcceptSocketAsync` and `TcpListener.AcceptTcpClientAsync` had overloads that took `CancellationToken`s.

What is the feasibility of this? I don't mind writing a proposal if I have to, I just am not familiar enough with .NET yet to write an effective one.

**Comment by stephentoub** [comment]
> What is the feasibility of this?

Trivial once Socket.AcceptAsync has an overload that supports cancellation, but while such an API has been approved, it's not yet implemented.

**Reply by JC3** [reply]
> but while such an API has been approved

Oh! So it's pretty much, nothing to do but cross my fingers and wait, right?

**Comment by stephentoub** [comment]
This was actually just implemented for .NET 6 in https://github.com/dotnet/runtime/pull/53340.
